Automattic/newspack-plugin
# fix(pwa): disable post request interception

### All Submissions:


### Changes proposed in this Pull Request:

This pull request resolves an issue in the WordPress PWA plugin where the service worker incorrectly handles WP REST API errors by intercepting requests and returning misleading HTTP statuses. By removing `wp-offline-post-request-handling`, REST API requests directly reach the server, ensuring accurate error propagation and improving reliability for developers and end-users.

### How to test the changes in this Pull Request:

1. Ensure PWA plugin is installed and active.
2. Navigate to `/wp-admin/admin.php?page=newspack-connections-wizard#/`
3. Open Network tab in Developer Tools
4. Disconnect MailChimp by clicking "Disconnect" to the right of the card.
5. Click Connect to open Modal.
6. Attempt connection with incorrect API key (i.e. force an error). See below:
![Screenshot 2024-05-31 at 09 26 58](https://github.com/Automattic/newspack-plugin/assets/3676975/455dd9fc-a481-40d4-b2bb-29150bcdc78c)
7. Click "Connect".
8. Observe in the Network tab that a fetch request has been made by the PWA service worker which has returned a 200 OK status code, however when previewing the response it is immediately clear that the request has resulted in a 500 server error derived from the MailChimp API package https://github.com/drewm/mailchimp-api/blob/master/src/MailChimp.php#L49.
![Screenshot 2024-05-31 at 09 18 17](https://github.com/Automattic/newspack-plugin/assets/3676975/6bd4360c-e37d-40ae-9296-6b5edbe19fc7)
10. Observe the error displayed is "The response is not a valid JSON response.", indicative that there's been an additional error in parsing the HTML response as JSON.
![Screenshot 2024-05-31 at 09 18 34](https://github.com/Automattic/newspack-plugin/assets/3676975/eac8d9db-3e87-470c-8bf6-5551afe4464c)

**Now to test fix:**

1. Checkout this branch.
2. Go back to the browser and clear all Application cache under Application tab in Developer Tools by clicking "Clear site data". Note: this action will log you out of wp-admin.
![Screenshot 2024-05-31 at 09 43 08](https://github.com/Automattic/newspack-plugin/assets/3676975/cf5289cc-8f2d-4896-9cfe-48951ff5abb4)
3. Refresh your browser, this will redirect to login, login and redirect back to Connections Wizard.
4. Click Connect to open Modal.
5. Again, attempt connection with incorrect API key.
6. This time you should see the correct error being displayed:
![Screenshot 2024-05-31 at 09 48 34](https://github.com/Automattic/newspack-plugin/assets/3676975/75da49ef-1d2e-4587-97f9-bdc638edac8f)

_Note: if you receive the same result as in test 1, attempt a hard refresh_

### Other information:
